QRCoder/QRCodeData.cs

```diff
@@ -121,60 +122,64 @@ public QRCodeData(byte[] rawData, Compression compressMode)
     /// <returns>Returns the raw data of the QR code as a byte array.</returns>
     public byte[] GetRawData(Compression compressMode)
     {
-        var bytes = new List<byte>();
-
-        //Add header - signature ("QRR")
-        bytes.AddRange(new byte[] { 0x51, 0x52, 0x52, 0x00 });
-
-        //Add header - rowsize
-        bytes.Add((byte)ModuleMatrix.Count);
+        using var output = new MemoryStream();
+        Stream targetStream = output;
+        DeflateStream? deflateStream = null;
+        GZipStream? gzipStream = null;
 
-        //Build data queue
-        var dataQueue = new Queue<int>();
-        foreach (var row in ModuleMatrix)
+        //Set up compression stream if needed
+        if (compressMode == Compression.Deflate)
         {
-            foreach (var module in row)
-            {
-                dataQueue.Enqueue((bool)module ? 1 : 0);
-            }
+            deflateStream = new DeflateStream(output, CompressionMode.Compress, true);
+            targetStream = deflateStream;
         }
-        for (int i = 0; i < 8 - (ModuleMatrix.Count * ModuleMatrix.Count) % 8; i++)
+        else if (compressMode == Compression.GZip)
         {
-            dataQueue.Enqueue(0);
+            gzipStream = new GZipStream(output, CompressionMode.Compress, true);
+            targetStream = gzipStream;
         }
 
-        //Process queue
-        while (dataQueue.Count > 0)
+        try
         {
-            byte b = 0;
-            for (int i = 7; i >= 0; i--)
+            //Add header - signature ("QRR")
+            targetStream.Write(new byte[] { 0x51, 0x52, 0x52, 0x00 }, 0, 4);
+
+            //Add header - rowsize
+            targetStream.WriteByte((byte)ModuleMatrix.Count);
+
+            //Build data queue
+            var dataQueue = new Queue<int>();
+            foreach (var row in ModuleMatrix)
             {
-                b += (byte)(dataQueue.Dequeue() << i);
+                foreach (var module in row)
+                {
+                    dataQueue.Enqueue((bool)module ? 1 : 0);
+                }
+            }
+            for (int i = 0; i < 8 - (ModuleMatrix.Count * ModuleMatrix.Count) % 8; i++)
```

Comment:
```suggestion
            int mod = (int)(((uint)ModuleMatrix.Count * (uint)ModuleMatrix.Count) % 8);
            for (int i = 0; i < 8 - mod; i++)
```
I'm not sure the JIT will keep the expression in a register. Re-evaluation is quite expensive, by multiplicaiton and especially the division (for the modulus).
So compute once as `uint`, as then the mod becomes a cheap bit-shift too.